capstone/src/main/java/com/google/sps/data/Lesson.java

```diff
@@ -95,6 +100,28 @@ public Date getDate() {
         return (Date) this.entity.getProperty(Lesson.DATE_PROPERTY_KEY);
     }
 
+    public static Lesson deserializeJson(String json) throws IOException {
+        JsonObject jobject = JsonParser.parseString(json).getAsJsonObject();
+
+        switch(jobject.get(Lesson.TYPE_PROPERTY_KEY).getAsString()) {
+            case Lesson.TYPE_FORM:
+                return new Form(jobject.get(Lesson.TITLE_PROPERTY_KEY).getAsString(), 
+                    jobject.get(Lesson.DESCRIPTION_PROPERTY_KEY).getAsString(), jobject.get(Form.EDIT_URL_PROPERTY_KEY).getAsString(),
+                    jobject.get(Form.URL_PROPERTY_KEY).getAsString());
```

Comment:
There is only the case for forms right now. I can add the cases for the other types of lessons (video, image, content) once I know how to do tests more efficiently.